### PR TITLE
Include more information in shed.yml

### PR DESCRIPTION
**Pull request by erasche**
We're still missing category IDs but we're one step closer to automatic repo creation.

I've synced the repo with the information currently available in the toolshed (some unicode characters were converted to their ascii counterparts).

Once category IDs are available in the toolshed (https://github.com/galaxyproject/tools-iuc/issues/73) we'll be able to automatically create ALL missing repositories.

### CODE REVIEW

**Comment by bgruening** [comment]
Who is this `Bjoern Gruening`? I think we should remove this guy from the description. I will clean this a little bit up before we merge this. Thanks @erasche for stating this!

**Comment by erasche** [comment]
sounds good @bgruening :)

Note that even after this PR is merged, the description/synopsis won't be synced to the toolshed. We have to wait on an API route for updating the repository with new information first.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/47/
Test (probably) passed

**Comment by martenson** [comment]
The way I perceived it the homepage_url is not meant to be the home of the Galaxy Tool but rather home of the Tool itself. E.g. http://130.235.46.10/ARAGORN/ for the aragorn package. What are your opinions about this?

**Comment by erasche** [comment]
@martenson I agree *wholeheartedly*! 

However, this is being automatically applied to ~175 files. Maybe I'll have time some other week to go through and determine homepages for all of these and make a new PR....

**Comment by peterjc** [comment]
In general I think the galaxy tool's home page is more useful - that's the first place to check if there are Galaxy packaging problems.

For the example you gave, https://toolshed.g2.bx.psu.edu/view/iuc/package_aragorn_1_2_36 and https://testtoolshed.g2.bx.psu.edu/view/iuc/package_aragorn_1_2_36 don't set a URL, but the detailed description says:

```
ARAGORN, tRNA (and tmRNA) detection.

http://www.ncbi.nlm.nih.gov/pubmed/14704338

Repository-Maintainer: Björn Grüning
Repository-Development: https://github.com/bgruening/galaxytools
```

(As an aside, @bgruening maybe that should use https://github.com/galaxyproject/tools-iuc/tree/master/packages/package_aragorn_1_2_36 now?)

**Comment by erasche** [comment]
@peterjc maybe we can make this clear to end users that they should check `remote_repository_url` before `homepage_url`?

**Comment by erasche** [comment]
And w/r/t aragorn, yeah, that pubmed ID should be a DOI somewhere. Maybe time to push for citations  on the package_*s rather than just on end-tools?

**Comment by peterjc** [comment]
Would https://github.com/galaxyproject/tools-iuc/tree/master/packages/package_aragorn_1_2_36  be acceptable as the ``remote_repository_url`` or should that be  https://github.com/galaxyproject/tools-iuc/ or even the git clone URL https://github.com/galaxyproject/tools-iuc.git instead (this last URL works fine in a browser)? For a human the first URL is more helpful as is drills down the relevant folder in what is now a very large repository, but isn't really a repository URL.

However, I still disagree about using ``homepage_url`` for the homepage of the underlying tool. I think it should be the URL of the Galaxy tool/package/wrapper itself.

For this example I might expect:

```
homepage_url = https://github.com/galaxyproject/tools-iuc/tree/master/packages/package_aragorn_1_2_36
remote_repository_url = https://github.com/galaxyproject/tools-iuc.git
```

Also, +1 on adding citation tags somehow to package definition XML files.

Should we be having this discussion on the galaxy-dev mailing list instead? CC @davebx @jmchilton

**Comment by peterjc** [comment]
(Just to be clear, I have no objections to merging this PR as it is with the URLs as harvested from the Tool Shed. I just think we need some clearer guidelines for what these URLs should be used for, which can be addressed in later commits/PRs)

**Comment by erasche** [comment]
Who needs a mailing list when you have git :P (apparently this is actually a thing, some groups get rid of their mailing list and just use github issues)

tl;dr +1 to your suggestion, if we mention upstream some how.

Long version:

Agreed, we definitely need a convention, and I agree with your assessment that having it pre-drilled down is very nice for end users, but on the other hand using it as a repo url that can be cloned (https, with or without the .git ending) seems like better practice based on the name/intention of that variable. 

Maybe this would imply the need for an "upstream_homepage" variable? I can see the argument for homepage_url being that of the galaxy tool, which works really well for your blast repositories since you include readmes with in all of your folders; having the upstream homepage point there is definitely good for end users. We still need to reference upstream, does that belong in another variable or in the description field?

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/48/
Test (probably) passed

**Comment by nsoranzo** [comment]
I'd like to note that there be more the 1 upstream tool used by a Galaxy tool.

**Comment by erasche** [comment]
@nsoranzo thanks, good to be aware of. Sounds more and more like something that should be added to the tool/package readme file.

**Comment by erasche** [comment]
@galaxyproject/iuc the IUC has many standards discussions like this, should we move them to a github repo or similar so we can keep our best practices under VC and discuss changes to them?

As it stands with them on the wiki, it seems a bit strange for me to garner the opinions of a couple people on a tangentially related github issue, and update the wiki as a result.

**Comment by martenson** [comment]
I had this in mind when designing these:

``remote_repository_url`` should be useful to developers and  is also meant to be a connection point for automatic fetch in the future, I would like it to point at the folder in which there is only one ``.shed.yml`` or similar direct way of connecting the repo in TS to dev repo.

``homepage_url`` should be useful to users that want to learn more about the tool, underlying paper and see the latest information and development about the tool.

**Comment by erasche** [comment]
@martenson but surely `remote_repository_url` won't be functional if pointing to a subdirectory of the web interface to a git repository? Surely it can *only* point at the base of the repo for automatic fetching, *unless* we move every tool into separate repos and start using submodules.

**Comment by martenson** [comment]
@erasche It can actually point to a subdirectory. We can use GitHub API to obtain everything we need in order to make the fetching possible.

**Comment by erasche** [comment]
@martenson and for non-github sources?

**Comment by martenson** [comment]
Github is the first to be adopted, others - if any - should be dealt when the need arises.

**Comment by erasche** [comment]
(was hoping for other IUC members to give some feedback, but I'll continue ahead)
@martenson okay, that works for me. That was the meaning I'd gleaned from the variable names. 

Do you see the remote_repo_url being something displayed to end users? I want to make sure @peterjc's concerns are addressed, that users go the correct place *first* for support/bugs in Galaxy wrappers before heading upstream.

If we can find an obvious way for the remote_repo_url to be used internally AND be visible/useful to end users as the primary place to go, then great, we can use that as the git subfolder. I worry that by doing that we're conflating their uses and if Galaxy ever supports something other than github we'll see trouble then.

**Comment by martenson** [comment]
That is a good idea. Let's leave this open for a day or two and let others chime in.

**Comment by bgruening** [comment]
If we really can add some more fields to `tool_dependencies.xml` here are my suggestions:

* add one citation field to every package section
* make the help section `rst` aware
* `remote_repository_url` should be automatically filled by planemo during upload and point to the directory where the `shed.yml` is located
* `homepage_url`: not sure this is useful if we have a citation tag inside `tool_dependencies.xml`, one idea would be to link to the tool developer homepage, e.g. IUC wiki page, group page ...

If we really want to have long descriptions in `shed.yml` I would like to see the long string description format. The following should work:

```
long_description: |
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Traditur, inquit, ab Epicuro ratio neglegendi 
  doloris. Haec quo modo conveniant, non sane intellego. Ait enim se, si uratur, Quam hoc suave! 
  dicturum. Qua igitur re ab deo vincitur, si aeternitate non vincitur? Sint ista Graecorum; Duo Reges: 
  constructio interrete. Ut in geometria, prima si dederis, danda sunt omnia. Ad eas enim res ab Epicuro 
  praecepta dantur.

  Istam voluptatem perpetuam quis potest praestare sapienti? His enim rebus detractis negat se reperire 
  in asotorum vita quod reprehendat. Nummus in Croesi divitiis obscuratur, pars est tamen divitiarum. 
  Cuius similitudine perspecta in formarum specie ac dignitate transitum est ad honestatem dictorum.
```

But I would like to see such information within the xml file. The only information that needs to be in `shed.yml` is imho:
 * owner: iuc
 * type: tool_dependency_definition 
 * name: package_meme_4_10_0 **
 * remote_repository_url: https://github.com/galaxyproject/tools-iuc/tools/some_folder/ ***

(** `name` could probably generated from the directory name 
*** `remote repository` could also be generated by planemo, maybe during `init`)

**Comment by erasche** [comment]
> add one citation field to every package section

+1

> make the help section rst aware

see below my comments on README/long_description (the help field)

> remote_repository_url should be automatically filled by planemo during upload and point to the directory where the shed.yml is located

not a fan of generating remote_repository_url with planemo, that seems like it'll be another hack[1]

> If we really want to have long descriptions in shed.yml I would like to see the long string description format. The following should work

+1, that's the format I use by hand. Pyyaml output the rubbish that you see and I don't know if I can fix that automatically without "manually" creating the yaml files (e.g. `output.write("name: %s")` rather than `output.write(yaml.dump())`). If you really want that done for all of them, I can go back and update the PR like that.

> name could probably generated from the directory name 

name can be removed everywhere in the IUC repo, but I like having it in there to be explicit[0]

> But I would like to see such information within the xml file

`long_description` could just go away if we consistently placed it in `README.rst`. We could just never fill it out. Would that work for you? If so I'll update by adding a readme everywhere. It would further be nice that when people were browsing our github, they'd see the nice READMEs. E.g. @peterjc's: https://github.com/peterjc/galaxy_blast/tree/master/tools/ncbi_blast_plus

> The only information that needs to be in shed.yml is imho

We'll need more than that I'm afraid, e.g. categories will be added (soon). `homepage_url` could be rolled into the `README.rst`, and it could be automated, but it seems like it goes back to the days of `Repository-Maintainer: Jane Doe` rather than using toolshed provided fields? 

[0] From a new-user perspective (or how I imagine that I would've reacted when I was a novice), I'd rather have where the repo name comes from to be very explicit, rather than it to be created from folder name via planemo. The IUC repos are supposed to be examples that people can copy. It's a small cost to be very clear about propagation of things like the name.
[1] I use the word hack, people will probably take issue with that, but I mean it in the sense of "planemo doing magic with git commands executed to find the origin, and then modifying that URL based on cwd" which seems like a LOT of "magic" to me... again things I wouldn't expect to happen as a new-user.

**Comment by bgruening** [comment]
> name can be removed everywhere in the IUC repo, but I like having it in there to be explicit[0]

`name` should definitely be there, but I think we can generate this field automatically if we follow some rules and we do most of the time already

> long_description could just go away if we consistently placed it in README.rst. We could just never fill it out. Would that work for you? If so I'll update by adding a readme everywhere. It would further be nice that when people were browsing our github, they'd see the nice READMEs. E.g. @peterjc's: https://github.com/peterjc/galaxy_blast/tree/master/tools/ncbi_blast_plus

Visibility of github repositories is an excellent point, but I'm more concert about the visibility of the Tool Shed repository. Imho all tool specific informations should go directly into the xml files and the Tool Shed should be smart about extracting these informations and creating a nice overview page. `homepage_url` can be in a readme file, fair point. Imho it would be more explicit and easier to extract in the xml file. (Only if we really need this information). For the readme file we need to distinguish between tool dependencies and tools. Tool dependencies currently can not have a readme file in the Tool Shed.

In the end it boils down to what we want to do with all these informations? Do we want to create a nice overview page for every tool, tool dependency and tool suite? Maybe even a overview page for every user? But this all very structured and from the design very similar? Than I would like to collect all informations in a structured way. Tool informations in xml, repository information in `shed.yml`. Otherwise, we can give everyone the freedom to design there own site with a readme file as we do currently for tool repositories.
 
> [0] From a new-user perspective (or how I imagine that I would've reacted when I was a novice), I'd rather have where the repo name comes from to be very explicit, rather than it to be created from folder name via planemo. The IUC repos are supposed to be examples that people can copy. It's a small cost to be very clear about propagation of things like the name.

I'm not talking about _not_ setting `name` in `shed.yml`, just saying that this value can be easily auto-generated. `name` should be definitely in `shed.yml`! 

> [1] I use the word hack, people will probably take issue with that, but I mean it in the sense of "planemo doing magic with git commands executed to find the origin, and then modifying that URL based on cwd" which seems like a LOT of "magic" to me... again things I wouldn't expect to happen as a new-user.

At some point we will have Tool Shed - github integration and this field will be populated automatically. I imagine the Tool Shed a lot like docker-hub in the end and here the remote_url is also set by default if you bind your repository to github.

As I see `planemo` currently: A wonderful tool to auto generate all boilerplate we currently need, this includes `shed.yml` and the user should be aware that all files needs to be checked and adopted: in the end they are just templates. We could leave the `remote_repository_url` empty or we can guess a good default.

**Comment by erasche** [comment]
> I'm not talking about not setting name in shed.yml, just saying that this value can be easily auto-generated. name should be definitely in shed.yml!

Ah, okay. Yes, definitely, *that* I agree with, planemo init could definitely do that. Sorry for the misunderstanding :)

> At some point we will have Tool Shed - github integration and this field will be populated automatically. I imagine the Tool Shed a lot like docker-hub in the end and here the remote_url is also set by default if you bind your repository to github.

Yep, pretty much, makes sense. As much as I love how easy things are with github I'm still a bit hesitant to tie developers to that saying "use this platform, it will make toolshed integration easier". I seem to be in the minority here and it's a *tiny* complaint, so I don't care that much. Github is the way to go then!

>  We could leave the remote_repository_url empty or we can guess a good default. 

Now that I know you mean planemo here, that comment makes a lot more sense :)

> Visibility of github repositories is an excellent point, but I'm more concert about the visibility of the Tool Shed repository. Imho all tool specific informations should go directly into the xml files and the Tool Shed should be smart about extracting these informations and creating a nice overview page. 

yep, sounds good so far

> homepage_url can be in a readme file, fair point. Imho it would be more explicit and easier to extract in the xml file. (Only if we really need this information). For the readme file we need to distinguish between tool dependencies and tools. Tool dependencies currently can not have a readme file in the Tool Shed. 

it seems weird to put it in an XML file when tool repositories don't have a single XML file. However, both tool/package repos have a single .shed.yml. 

As I see it a README and long_description serve the same functionality. It always felt like a duplicate field. If we phased out using long_description and pushed use of a README.rst (would solve your comment about wanting the long_description to support rST), this would be an acceptable solution here? If so, I can probably manage implementing that.

> In the end it boils down to what we want to do with all these informations? Do we want to create a nice overview page for every tool, tool dependency and tool suite? Maybe even a overview page for every user? But this all very structured and from the design very similar? Than I would like to collect all informations in a structured way. Tool informations in xml, repository information in shed.yml. Otherwise, we can give everyone the freedom to design there own site with a readme file as we do currently for tool repositories.

I'd argue that we want a nice overview page for every repository, regardless of the repo type; meaning that that this is repository level metadata and belongs in the .shed.yml. I like the freedom of designing their own README, as it would provide a nice repo-level overview in **both** github and toolshed.

**Comment by peterjc** [comment]
I like this plan:

*As I see it a README and long_description serve the same functionality. It always felt like a duplicate field. If we phased out using long_description and pushed use of a README.rst (would solve your comment about wanting the long_description to support rST), this would be an acceptable solution here?*

The only problem is tool dependency type repositories currently insist on exactly the one XML file. The Tool Shed would need to also allow a readme file (insisting on ``README.rst`` would be nice but inconsistent with the more relaxed README naming for ordinary tools on the Tool Shed).

**Comment by erasche** [comment]
I'm sure that'll be possible to relax that requirement and permit readmes (even perhaps displaying the readme from long_description if no readme file is found)? @martenson thoughts? I'm happy to make the PR for it.

**Comment by jmchilton** [comment]
I don't agree with all of this - but I am definitely a huge :+1: on @erasche opening a PR to relax the requirement on those repositories to allow READMEs :). I vaguely remember looking at this and thinking it would be harder than it would seem to implement.

**Comment by erasche** [comment]
@jmchilton if you have thoughts, please express them! We're open to feedback before the community rushes off to implement things.

**Comment by erasche** [comment]
@bgruening okay, ready(ish) for review, if you would mind looking over it :)

I've added categories, but in the diff it looks like a lot more came along than expected, which I didn't expect. Let me know if you want me to re-do something!

**Comment by bgruening** [comment]
@erasche will do. Please update me, did we agreed to store everything in `shed.yml` and remove the `help` section from `tool_dependencies.xml` files?

**Comment by erasche** [comment]
@bgruening I believe opinion was that removing help and moving everything to `.shed.yml` wasn't as good of a solution as supporting `README(.rst)` files for package repositories. Then we could use those (and rST) instead of `long_description`.

Since that will require some dev work I didn't want to add that as a dependency of #73 and cause this to delay further.

**Comment by erasche** [comment]
I've come to the conclusion that I'm solidly in favour of using `homepage_url` for the upstream home. Here's my rationale if anyone would like to comment.

- We need to link to upstream
  - upstream is what's important, not our tiny XML/python/shell script wrappers that shoe-horn their software into galaxy.
  - Thus it seems weird to me to be using `homepage_url` to point to the wrapper, when users hear "homepage" and might not understand that there's a shim that we've written to get tool X into Galaxy.
- I can understand reasoning for wanting to point at the github *tool folder*, as users may want to contribute to the wrapper.
- My conclusion is that we need to *display* a clone address.
  - Bjoern has been doing this for a while with `Repository-Maintainer/Repository-Development`. 
  - If we're pointing at github, we have commit logs available and `-Maintainer` can go away. However, we still need to point at the repo itself.
  - That's what `remote_repository_url` is for (i.e. the github/development)
  - We can address @peterjc's issue of pointing to the clone address by simply displaying that URL (calculated from the `remote_repository_url`). I like the idea that users can see the upstream repo and clone from it without leaving the toolshed.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/63/
Test (probably) failed.

**Comment by jmchilton** [comment]
Slipping back into [this crazed and unpopular guy](https://github.com/galaxyproject/tools-devteam/pull/16#issuecomment-60269743) - I don't agree with your characterization of Galaxy tool development - I feel it belittles tools and tool authors. I don't think we should assume that homepage is some single website for a single application that a tool is wrapping - this is not the mental model we should apply to tools even if they are the most common type of tool. For this reason ``homepage_url`` in my mind should be the homepage for the tool - the thing that perhaps people don't care about yet but which we should be trying to make them care about - and which may correspond to no applications, 400 applications, a web service, etc....

I have no problem with generally being able to link out to resources corresponding to requirements of a tool - but it shouldn't be a single URL - it should be a **list** of websites and/or citations (DOIs) then.

**Comment by erasche** [comment]
@jmchilton you raise fair points there :) and if it were as "easy" as writing shims, then it wouldn't be of value to publications, I believe the scientific community has already passed comment on that by accepting many tool-based papers for publication
 
tl;dr: +1 to `homepage_urls` (**plural**), but I don't want to keep sticking URLs/DOIs in `long_description` because it's not machine parseable and the problem is just going to get worse. I want the situation to improve even if it's not getting to perfect.

Hadn't seen that comment you linked before, and I'd disagree that it's an unpopular opinion (heck you're posting to the group of extremely avid tool devs). But here, just for you :P

kek

I have a bit of trouble with where the line is drawn in the sand for tools vs collections of tools (possible multiple upstreams, but should REALLY be a `suite`) vs "here is a utility which abstracts a LOT of underlying things and you don't need to worry about them" (i.e. the case that comes to my mind when people say "multiple homepages").

I've always *liked* the idea that Galaxy tools aren't *just* simple wrappers, that there would be a single "FastQ to FastA" tool, and not half a dozen versions of those which are all provided by different toolkits various people have wrapped (c.f. me not helping the problem by wrapping of `seqtk`). 

However, I haven't seen that as what happens in practice. What I see (and please provide counterpoints here if your experience differs) is that people wrap specific toolkits to provide the toolkit more than the functionality (getting off into the sticks: I imagine the "brand recognition" helps visibility of tools there to some extent. Who do you trust more, "Bedtools bed2gff3" or "bed to gff3 converter", and yes, ratings would help here/etc.)

Back to the point, while I don't want to decrease the visibility of Galaxy tool authors, I think that since many people wrap specific upstream packages, the current practice should be that `homepage_url` is an upstream homepage, **UNTIL** `homepage_url` can be made into a list. I will concede that, it would be *awesome* for it to support multiple tools, and would make this problem go away, to my understanding.

**Comment by erasche** [comment]
If there's an unpopular opinion there, it's my offhand (and regrettable) comment about the value placed on tools vs upstream. I said that intending to note the orders of magnitude of difference in amount of energy that went into developing upstream, and wrapping it for use in Galaxy. I want to ensure that upstreams get highlighted. `homepage_url` seems like the place to do that, rather than just in a `long_description` or in a README for the package/tool on github. Lastly, I'd rather have it as structured data than more unstructured content.

**Comment by peterjc** [comment]
There is a continuum here. For "simple" wrappers, I've come to agree with @erasche  your point that ``homepage_url`` ought to be the upstream tool (provided the wrapper's repository is also prominent).

For complex tools in their own right, calling one or more upstream tools, then no - the synthesis is a novel tool, and deserves the primary home page promotion. e.g. I would take this stance for my ``blast_rbh`` https://github.com/peterjc/galaxy_blast/tree/master/tools/blast_rbh even though it calls NCBI BLAST+ internally.

If we agree with this, we should write something along these lines in the Best Practice guidelines.

**Comment by nsoranzo** [comment]
To put it simple, I agree with @jmchilton .

**Comment by erasche** [comment]
@peterjc I can *definitely* agree with that, there is a continuum, and I'd forgotten about tools which are developed in their own right as galaxy tools. (Despite the fact that this make up 90% of the stuff we work on internally and intend to publish someday).

So, working thoughts:

Case | `homepage_url` | notes
--- | --- | ---
package of upstream | upstream's home | 
simple wrapper tool for package of upstream | upstream's home | repository_url will be used to get to the XML development
complex wrapper (e.g. with .py) for package of upstream | `repository_url == homepage_url` ? | E.g. @peterjc's case with BlastRBH. Here package's URL could be mentioned in README (hopefully conflated with long_description), or there could be better viz of package_* which mentions upstream. Additionally dependency `homepage_url`s could be appended to the tool's `homepage_url`

Action items:

- `homepage_url` turned into a list
- DOIs for packages included in citations

**Comment by erasche** [comment]
[spoiler] This is why I suggested moving discussion to a best practices repo[/spoiler]

> the IUC has many standards discussions like this, should we move them to a github repo or similar so we can keep our best practices under VC and discuss changes to them?

**Comment by nsoranzo** [comment]
@erasche Why not used something like `dependencies_url` (a list) and leave `homepage_url` for the wrapper homepage? I would find it much less confusing.

> [spoiler] This is why I suggested moving discussion to a best practices repo[/spoiler]

:+1: So we can also have issues/pull requests...

**Comment by erasche** [comment]
@nsoranzo the thinking there was that we wanted to support multiple homed projects, e.g. @peterjc's BlastRBH. He wrote custom code which should probably be the homepage for the tool, but we still wanted to mention the upstream blast tool that supports it somehow. 

Maybe my suggestion of 'appending `homepage_url`s from dependencies was a bit extreme.'. 

I've opened a repository for this and I'll start migrating standards there today. Anyone wanting to continue this discussion, please comment on galaxy-iuc/standards#1

File: packages/package_ccat_3_0/.shed.yml [hunk 1]
```diff
@@ -1,3 +1,11 @@
 # repository published to https://toolshed.g2.bx.psu.edu/repos/iuc/package_ccat_3_0
 owner: iuc
 name: package_ccat_3_0
+type: tool_dependency_definition
+description: |
+    CCAT is a software package for the analysis of ChIP-seq data with negative
+    control.
+homepage_url: https://github.com/galaxyproject/tools-iuc/
+long_description: remote_repository_url: https://github.com/galaxyproject/tools-iuc/
```
Comment (nsoranzo) [inline packages/package_ccat_3_0/.shed.yml]: `long_description` is empty here, I don't know if it's ok.

For sure `remote_repository_url` should go on a separate line.

Comment (erasche) [inline packages/package_ccat_3_0/.shed.yml]: Oh wow, nice catch. Thanks.

Comment (erasche) [inline packages/package_ccat_3_0/.shed.yml]: Empty long description is fine.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/84/
Test (probably) passed

**Comment by bgruening** [comment]
@erasche how to you want to me to help here. Should I commit to your branch or should I merge this and fix afterwards? As it seems it's mainly yaml fixes to use '|' and remove `repository_maintainer` and `remote_repository_url`.

**Comment by erasche** [comment]
Feel free to commit to the branch!
31. mar. 2015 06.07 skrev "Björn Grüning" <notifications@github.com>:

**Comment by erasche** [comment]
@bgruening I'm automating some of your requested fixes, I'll ping you again when it's ready. Thanks again :)

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/87/
Test (probably) passed

**Comment by erasche** [comment]
Okay, had to do a bad thing and force push due to a rebase where I squashed an extra commit than I'd intended to. Oh well. @bgruening this is ready for review.

- `homepage_url` removed
- `remote_repository_url` updated per spec and @martenson (now points to folder with .shed.yml file)
- long_description reflowed to use `long_desc: | ...` style.

File: packages/package_aragorn_1_2_36/.shed.yml [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
+categories:
+- Tool Dependency Packages
+description: Contains a tool dependency definition that downloads and extract version
+  1.2.36 of Aragorn.
+long_description: |
+  ARAGORN, tRNA (and tmRNA) detection.
+
+  http://www.ncbi.nlm.nih.gov/pubmed/14704338name: package_aragorn_1_2_36
 owner: iuc
-name: package_aragorn_1_2_36
```
Comment (martenson) [inline packages/package_aragorn_1_2_36/.shed.yml]: this guy lost its name

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/88/
Test (probably) passed

File: packages/package_aragorn_1_2_36/.shed.yml (hunk 2 above)
Comment (erasche) [inline packages/package_aragorn_1_2_36/.shed.yml]: crap, you're right. Nice catch. Will grep for more of the same.

Comment (erasche) [inline packages/package_aragorn_1_2_36/.shed.yml]: Okay, that looks like it's the only one, unless my regex fails me.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/89/
Test (probably) passed

**Comment by erasche** [comment]
@bgruening is this okay to merge? It's been open 20 days, and final fixes were in as of 5 days ago. Do you still have any concerns? Otherwise I'd like to get packages pushed to TTS.

**Comment by martenson** [comment]
What happens if I change some of the parameters in the TS directly but not in the .shed.yml? Which version will be treated as latest?

**Comment by erasche** [comment]
`.shed.yml` updates will consistently overwrite whatever is in the TS. 

It may be my SysAdmin background, but personally I somewhat prefer this as it means that all changes to package names/descriptions/categories are accounted for and attributable to a specific individual within the IUC.

**Comment by martenson** [comment]
Don't get me wrong - I am :+1: on this PR - but this may create usability issues e.g. with categories - you have to use the UI or API to get the list of available categories but then you have to manually enter and maintain it in .shed.yml because there is no other way to sync. Also other tool devs may have different setup than the IUC has regarding the workflow.
 
I like the approach to treat TS as a blackbox API however I wanted to throw this out there so we are aware.

**Comment by erasche** [comment]
True, you do have to manually get a list of categories. I'm not a fan of that either. 

I have some very ... rough scripts to automatically pull information from a TS and stick into a `.shed.yml` which may ease the process for other groups wanting to adopt our practices. I definitely agree that finding the list of available categories manually is unpleasant and should be eased. 

That said, it *should* be as simple as going to the [api route](https://toolshed.g2.bx.psu.edu/api/categories/) or [home page](https://toolshed.g2.bx.psu.edu/) and finding the correct string.

Just throwing ideas out, here's one possible solution, I could write up a script (for jenkins) that would run the "update info from TS" tool, and then if any changes were detected in the data, it'd open a PR for review. The group could merge that into master as needed, allowing a semi-manual version of bidirectional syncing.

File: packages/package_gnuplot-py_1.8/.shed.yml [hunk 3]
```diff
@@ -1,3 +1,17 @@
...
+  Gnuplot.py is a Python package that interfaces to gnuplot, the popular
+  open-source plotting program. It allows you to use gnuplot from within Python
+  to plot arrays of data from memory, data files, or mathematical functions. If
+  you use Python to perform computations or as "glue" for numerical programs,
+  you can use this package to plot data on the fly as they are computed. And
+  the combination with Python makes it is easy to automate things, including to
+  create crude "animations" by plotting different datasets one after another.
+
+  http://gnuplot-py.sourceforge.net/
 name: package_gnuplot-py_1.8
```
Comment (bgruening) [inline packages/package_gnuplot-py_1.8/.shed.yml]: Is `-` really allowed as part of a repository in the Tool Shed?

Comment (erasche) [inline packages/package_gnuplot-py_1.8/.shed.yml]: 1. No
2. The repo isn't even published to MTS, nor TTS, despite the "repository published ..." statement from the top
3. Thanks again for going over this @bgruening. I owe you a couple of beverages of your choice for this.

Comment (erasche) [inline packages/package_gnuplot-py_1.8/.shed.yml]: 4. It looks like I can blame myself for adding the line about "repository published to...", that was from the automated addition of `.shed.yml`s for all repositories.

Comment (martenson) [inline packages/package_gnuplot-py_1.8/.shed.yml]: VALID_REPOSITORYNAME_RE = re.compile( "^[a-z0-9\_]+$" )

Comment (erasche) [inline packages/package_gnuplot-py_1.8/.shed.yml]: Oy, lovely. I'll remove the period too.

Comment (erasche) [inline packages/package_gnuplot-py_1.8/.shed.yml]: Done.

**Comment by bgruening** [comment]
@martenson Is it ok to merge? I have a list of post-merge changes I will work on. We can discuss how to proceed with all these information during our next IUC meeting?

**Comment by bgruening** [comment]
@erasche nice work, most of the packages are converted successfully. I'm ready to merge and have a list of post-merge changes.

**Comment by erasche** [comment]
@bgruening great, thank you!

**Comment by martenson** [comment]
looks good, thanks guys!

**Comment by erasche** [comment]
Thank you everyone for your input. I've flipped the `--force_repository_creation` flag on planemo, so we should be creating new repositories as we push them now.

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/93/
Test (probably) passed

**Comment by erasche** [comment]
Refer to this link for build results (access rights to CI server needed): 
http://gx.hx42.org//job/IUC-Github/95/
Test (probably) passed
